### PR TITLE
Deployment api usage

### PR DESCRIPTION
**Pull request by shogunpurple**
## Description
Addresses the builder part of https://github.com/Budibase/budibase/issues/673

This PR contains code for user deployments to update their usage quotas - it relies heavily on the lambda code in https://github.com/Budibase/budibase-infra/pull/22.

### CODE REVIEW

File: packages/server/src/api/controllers/deploy/aws.js [hunk 1]
```diff
@@ -22,11 +22,42 @@ async function invalidateCDN(cfDistribution, appId) {
     .promise()
 }
 
-exports.fetchTemporaryCredentials = async function() {
+exports.updateDeploymentQuota = async function(quota) {
+  const DEPLOYMENT_SUCCESS_URL =
+    process.env.DEPLOYMENT_CREDENTIALS_URL + "deploy/success"
```
Comment (mike12345567) [inline packages/server/src/api/controllers/deploy/aws.js:27]: I've been trying to pass all new environment variables through the `environment.js` file, purely again so its all centralised, I think if we manage it that way going forward it'll be much easier for self hosting to work out what is needed.

File: packages/server/src/api/controllers/deploy/index.js [hunk 2]
```diff
@@ -31,13 +35,40 @@ async function replicateCouch({ instanceId, clientId, credentials }) {
...
+  const records = await db.allDocs({
+    startkey: "record:",
+    endkey: `record:\ufff0`,
+  })
+  const existingRecords = records.rows.length
+
+  const designDoc = await db.get("_design/database")
+
+  return {
+    records: existingRecords,
```
Comment (mike12345567) [inline packages/server/src/api/controllers/deploy/index.js:49]: Just a question, in terms of local storage and users created, do we want to count them as well?

**Comment by mike12345567** [comment]
LGTM -> just a few questions!

Also my commit was just merging in master!

File: packages/server/src/api/controllers/deploy/index.js (hunk 2 above)
Comment (shogunpurple) [reply packages/server/src/api/controllers/deploy/index.js:49]: Yep, we do for users, but the storage will be incremented when we upload things based on your PR, right?

Comment (mike12345567) [reply packages/server/src/api/controllers/deploy/index.js:58]: true yes, sorry I didn't understand that the `upload` endpoint would be used for when the storage is send up, that makes sense in that case!

Comment (shogunpurple) [reply packages/server/src/api/controllers/deploy/index.js:58]: Updated for users 👍

**Pull request merged**